pkg/apis/fission.io/v1/validation.go

```diff
@@ -38,26 +38,28 @@ var (
 	validAzureQueueName = regexp.MustCompile("^[a-z0-9][a-z0-9\\-]*[a-z0-9]$")
 )
 
-type ValidationErrorType int
+type (
+	ValidationErrorType int
 
-// ValidationError is a custom error type for resource validation.
-// It indicate which field is invalid or illegal in the fission resource.
-// Also, it shows what kind of error type, bad value and detail error messages.
-type ValidationError struct {
-	// Type of validation error.
-	// It indicates what kind of error of field in error output.
-	Type ValidationErrorType
+	// ValidationError is a custom error type for resource validation.
+	// It indicate which field is invalid or illegal in the fission resource.
+	// Also, it shows what kind of error type, bad value and detail error messages.
+	ValidationError struct {
+		// Type of validation error.
+		// It indicates what kind of error of field in error output.
+		Type ValidationErrorType
 
-	// Name of error field.
-	// Example: FunctionReference.Name
-	Field string
+		// Name of error field.
+		// Example: FunctionReference.Name
+		Field string
 
-	// Error field value.
-	BadValue interface{}
+		// Error field value.
+		BadValue string
```

Comment:
Change from the interface to string for code generation. Not found a better way to solve the code generation problem when field type is interface.